localstack-core/localstack/services/lambda_/api_utils.py

```diff
@@ -210,25 +278,57 @@ def get_name_and_qualifier(
     function_arn_or_name: str, qualifier: str | None, context: RequestContext
 ) -> tuple[str, str | None]:
     """
-    Takes a full or partial arn, or a name and a qualifier
-    Will raise exception if a qualified arn is provided and the qualifier does not match (but is given)
+    Takes a full or partial arn, or a name and a qualifier.
 
     :param function_arn_or_name: Given arn (or name)
     :param qualifier: A qualifier for the function (or None)
     :param context: Request context
     :return: tuple with (name, qualifier). Qualifier is none if missing
+    :raises: `ResourceNotFoundException` when the context's region differs from the ARN's region
+    :raises: `AccessDeniedException` when the context's account ID differs from the ARN's account ID
+    :raises: `ValidationExcpetion` when a function ARN/name or qualifier fails validation checks
+    :raises: `InvalidParameterValueException` when a qualified arn is provided and the qualifier does not match (but is given)
     """
-    function_name, arn_qualifier, _, arn_region = function_locators_from_arn(function_arn_or_name)
+    function_name, arn_qualifier, account, region = function_locators_from_arn(function_arn_or_name)
+    operation_type = context.operation.name
+
+    if operation_type not in _supported_resource_based_operations:
+        if account and account != context.account_id:
+            raise AccessDeniedException(None)
+
+    # TODO: should this only run if operation type is unsupported?
+    if region and region != context.region:
+        raise ResourceNotFoundException(
+            f"Functions from '{region}' are not reachable in this region ('{context.region}')",
+            Type="User",
+        )
+
+    validation_errors = []
+    if function_name:
+        validation_errors.extend(validate_function_name(function_arn_or_name, operation_type))
```

Comment:
I think we are still missing cases here. If the function name is `None`, which is possible when you specify an completely invalid value for it, for example `*`, the checks never catch it, and it continues with the operation, leading to errors down the line. AWS validation catches that.

Reply:
Removed that `None` check and added a case for this